examples/async-sign/signed_message/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+#![no_std]
+
+use codec::Decode;
+use gstd::{debug, exec, msg, prelude::*, ProgramId};
+use scale_info::TypeInfo;
+use sp_core::{
+    crypto::{AccountId32, UncheckedFrom},
+    sr25519,
+};
+
+#[derive(Debug, Decode, TypeInfo)]
+struct InitArgs {
+    account: AccountId32,
+}
+
+#[derive(Debug, Decode, TypeInfo)]
+pub struct HandleArgs {
+    pub message: Vec<u8>,
+    pub signature: Vec<u8>,
+}
+
+gstd::metadata! {
+    title: "SignedMessage",
+        init:
+            input: InitArgs,
+        handle:
+            input: HandleArgs,
+}
+
+static mut SIGNATORY: Option<ProgramId> = None;
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let signatory = match SIGNATORY {
+        None => {
+            msg::reply_bytes(b"Uninitialized", 10_000, 0);
+            return;
+        }
+        Some(s) => s.0,
+    };
+
+    let args: HandleArgs = match msg::load() {
+        Err(_) => {
+            msg::reply_bytes(b"Failed to decode `HandleArgs`", 10_000, 0);
+            return;
+        }
+        Ok(a) => a,
+    };
+    debug!("{:?}", args);
+
+    // the same way as in verify.rs from subkey
+    let mut signature = <sr25519::Pair as sp_core::Pair>::Signature::default();
+    if args.signature.len() != AsRef::<[u8]>::as_ref(&signature).len() {
+        msg::reply_bytes(b"Wrong signature length", 10_000, 0);
+        return;
+    }
+
+    signature.as_mut().copy_from_slice(&args.signature);
+
+    let pub_key = <sr25519::Pair as sp_core::Pair>::Public::unchecked_from(signatory);
```

Comment:
Compiler complains that `sp_core::Pair` is not in scope and I resolved that in this way. However it doesn't matter anymore since the code rewritten after other remarks